src/fields/field_fe.cc

```diff
@@ -290,11 +310,132 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
+	flags_ = init_data.flags_;
+
+
+	// read mesh, create tree
+    {
+       reader_file_ = FilePath( rec.val<FilePath>("mesh_data_file") );
+       auto reader_data = ReaderInstance::get_instance(reader_file_);
+       source_mesh_ = reader_data.mesh_;
+       source_mesh_->get_bih_tree(); // only create BIH tree
+	   // no call to mesh->setup_topology, we need only elements, no connectivity
+    }
+
+	field_name_ = rec.val<std::string>("field_name");
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::set_mesh(const Mesh *mesh, bool boundary_domain) {
+	// Mesh can be set only for field initialized from input.
+	if ( flags_.match(FieldFlag::equation_input) && flags_.match(FieldFlag::declare_input) ) {
+		ASSERT(field_name_ != "").error("Uninitialized FieldFE, did you call init_from_input()?\n");
+		this->make_dof_handler(mesh);
+	}
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::make_dof_handler(const Mesh *mesh) {
+	// temporary solution - these objects will be set through FieldCommon
+	map1_ = new MappingP1<1,3>();
+	map2_ = new MappingP1<2,3>();
+	map3_ = new MappingP1<3,3>();
+	fe1_ = new FE_P_disc<0,1,3>();
+	fe2_ = new FE_P_disc<0,2,3>();
+	fe3_ = new FE_P_disc<0,3,3>();
+
+	dh_ = new DOFHandlerMultiDim( const_cast<Mesh &>(*mesh) );
+	dh_->distribute_dofs(*fe1_, *fe2_, *fe3_);
+    unsigned int ndofs = max(dh_->fe<1>()->n_dofs(), max(dh_->fe<2>()->n_dofs(), dh_->fe<3>()->n_dofs()));
+    dof_indices = new unsigned int[ndofs];
+
+    // allocate data_vec_
+	unsigned int data_size = dh_->n_global_dofs();
+	data_vec_ = new VectorSeqDouble();
+	data_vec_->resize(data_size);
+}
+
+
+
+template <int spacedim, class Value>
+bool FieldFE<spacedim, Value>::set_time(const TimeStep &time) {
+	// Time can be set only for field initialized from input.
+	if ( flags_.match(FieldFlag::equation_input) && flags_.match(FieldFlag::declare_input) ) {
+	    ASSERT(field_name_ != "").error("Uninitialized FieldFE, did you call init_from_input()?\n");
+		ASSERT_PTR(dh_)(field_name_).error("Null target mesh pointer of finite element field, did you call set_mesh()?\n");
+		if ( reader_file_ == FilePath() ) return false;
+
+		// value of last computed element must be recalculated if time is changed
+		computed_elm_idx_ = numeric_limits<unsigned int>::max();
+
+		GMSH_DataHeader search_header;
+		search_header.actual = false;
+		search_header.field_name = field_name_;
+		search_header.n_components = this->value_.n_rows() * this->value_.n_cols();
+		search_header.n_entities = source_mesh_->element.size();
+		search_header.time = time.end();
+
+		bool boundary_domain_ = false;
+		std::vector<double> data_vec = *(ReaderInstances::instance()->get_reader(reader_file_)->template get_element_data<double>(search_header,
+				source_mesh_->elements_id_maps(boundary_domain_), this->component_idx_));
+		std::vector<double> sum_val(4);
+		std::vector<unsigned int> elem_count(4);
+
+		FOR_ELEMENTS( dh_->mesh(), ele ) {
```

Comment:
Make separate (private) method for the interpolation. Further we shall probably make separate generic class from that for interpolation between any fields. An then we can have different generic interpolation algorithms.